java/vector/src/main/java/org/apache/arrow/vector/ExtensionTypeVector.java

```diff
@@ -130,6 +130,12 @@ public TransferPair makeTransferPair(ValueVector target) {
     return underlyingVector.makeTransferPair(target);
   }
 
+  @Override
+  protected Class<? extends FieldReader> getReaderImplClass() {
+    throw new UnsupportedOperationException("Readers for extension types depend on the underlying vector, " +
```

Comment:
Throwing an exception here because `getReader` is delegating to the underlying vector, which to my understanding would always be a subclass of `BaseValueVector`. This means that there'd be a concrete implementation of `getReaderImplClass` that'd be available, so delegating to the `getReader` method of the underlying vector is fine and there's no need of an implementation here.

Please let me know if this is not the case.

Reply:
That should be fine I think -- should I add a comment explicitly saying that?

Reply:
Hmm, on second thought: maybe remove the override for getReader here, and put the delegation here instead? That way, users can override this and get the expected behavior. Otherwise they have to override both methods which is a bit of a mess.